File: fibonnacci/c/fibonacci.c
#include <stdio.h>
#include <stdlib.h>

unsigned long fibonacci(unsigned long n) {

	unsigned long n1, n2;
	unsigned long sum;

	if (0 == n)
		return 0;
	else if (n >= 1 && n <= 2)
		return 1;

	// starting from 3rd
	// 0, 1, 1, ...

	n -= 2;

	n1 = 1;
	n2 = 1;

	while(n--) {
		sum = n1 + n2;
		n1 = n2;
		n2 = sum;
	}

	return sum;
}

int main(int argc, char* argv[]) {

	unsigned long n;

	if (2 != argc) {
		fprintf(stderr, "usage: %s <n>\n", argv[0]);
		return 1;
	}

	n = atoi(argv[1]);

	n = fibonacci(n);

	printf("%li\n", n);

	return 0;
}

File: fibonnacci/python2/fibonacci.py
#!/usr/bin/python2

import sys

if 2 != len(sys.argv):
    sys.stderr.write("usage: " + sys.argv[0] + " <n>\n")
    sys.exit(1)

n = int(sys.argv[1])

def fibonacci(n):
    if (0 == n):
        return 0
    elif (n >= 1 and n <= 2):
        return 1

    n -= 2

    s = 0  # sum
    n1 = 1
    n2 = 1
    while (0 != n):
        s = n1 + n2
        n1 = n2
        n2 = s

        n -= 1

    return s

n = fibonacci(n)

print(n)
